### PR TITLE
Convert `highlighter` to TS

### PR DESCRIPTION
**Pull request by lyzadanger**
Prepare for some upcoming changes to this module.

### CODE REVIEW

File: src/annotator/highlighter.ts [hunk 1]
```diff
@@ -54,7 +57,10 @@ function getPDFCanvas(highlightEl) {
  *   above the PDF.
  * @param {string} [cssClass] - CSS class(es) to add to the SVG highlight elements
  */
-function drawHighlightsAbovePDFCanvas(highlightEls, cssClass) {
+function drawHighlightsAbovePDFCanvas(
```
Comment (robertknight) [inline src/annotator/highlighter.ts:60]: The `@param`s above can be simplified by removing the types.

File: src/annotator/highlighter.ts [hunk 2]
```diff
@@ -124,37 +129,22 @@ function drawHighlightsAbovePDFCanvas(highlightEls, cssClass) {
...
   if (range.collapsed) {
     // Exit early for an empty range to avoid an edge case that breaks the algorithm
     // below. Splitting a text node at the start of an empty range can leave the
     // range ending in the left part rather than the right part.
     return [];
   }
 
-  /** @type {Node|null} */
-  let root = range.commonAncestorContainer;
-  if (root.nodeType !== Node.ELEMENT_NODE) {
```
Comment (robertknight) [inline src/annotator/highlighter.ts:157]: Huh. I wonder why this didn't error before due to `root` being possibly-null.

File: src/annotator/highlighter.ts [hunk 3]
```diff
@@ -208,7 +196,10 @@ function wholeTextNodesInRange(range) {
  * @param {string} [cssClass] - CSS class(es) to add to the highlight elements
  * @return {HighlightElement[]} - Elements wrapping text in `normedRange` to add a highlight effect
  */
-export function highlightRange(range, cssClass) {
+export function highlightRange(
```
Comment (robertknight) [inline src/annotator/highlighter.ts:199]: The types can be removed from the `@param` and `@return` above.

File: src/annotator/highlighter.ts [hunk 4]
```diff
@@ -339,7 +322,7 @@ export function setHighlightsFocused(highlights, focused) {
       // SVG elements are rendered in document order so to achieve this we need
       // to move the element to be the last child of its parent.
       if (focused) {
-        const parent = /** @type {SVGElement} */ (h.svgHighlight.parentNode);
+        const parent = h.svgHighlight.parentNode as SVGSVGElement;
```
Comment (robertknight) [inline src/annotator/highlighter.ts:325]: The type was changed here from `SVGElement` (ie. _any_ SVG element) to `SVGSVGElement` (ie. specifically the `<svg>` element). Was that intentional? If we don't need to be that specific, I'd prefer to use the more general type.

**Review by robertknight — APPROVED** [review]
LGTM with some minor comments and corrections.

File: src/annotator/highlighter.ts (hunk 1 above)
Comment (lyzadanger) [reply src/annotator/highlighter.ts:60]: Ah! Yes, that's a nice idea.

File: src/annotator/highlighter.ts (hunk 4 above)
Comment (lyzadanger) [reply src/annotator/highlighter.ts:325]: Ah, nice catch.

**Pull request merged**